app/controllers/application_controller.rb

```diff
@@ -12,15 +12,19 @@ def current_user
     @user ||= User.find(session[:user_id]) if session[:user_id]
   end
 
-  def authorized_user!
-     redirect_to admin_path if current_admin?
+  def admin_authenticate
+    if current_user.role == 0
```

Comment:
We can probably refactor this later but for now it does the thing and that's awesome!!!